Classes/Controller/GetFileController.php

```diff
@@ -95,48 +99,74 @@ public function attachmentAction()
 
                     header('Content-Type: text/xml; charset=UTF-8');
 
-                }
+                    return $metsXml;
+
+                } else {
+
+                    http_response_code(404);
 
-                return $metsXml;
+                    print('No such document');
+
+                    exit;
+
+                }
 
             case 'attachment':
                 $path = rtrim('http://' . $fedoraHost, "/") . '/fedora/objects/' . $piVars['qid'] . '/datastreams/' . $piVars['attachment'] . '/content';
                 break;
+
             default:
-                break;
+                http_response_code(404);
+                print('No such action');
```

Comment:
Turns out `return $somestring` sets the response content respectively. Empty return results in HTML rendered TYPO3 error message.